pkg/pattern/drain/chunk_test.go

```diff
@@ -64,9 +64,9 @@ func TestForRange(t *testing.T) {
 		{
 			name: "No Overlap",
 			c: &Chunk{Samples: []logproto.PatternSample{
-				{Timestamp: 1, Value: 2},
-				{Timestamp: 3, Value: 4},
-				{Timestamp: 5, Value: 6},
+				{Timestamp: 2, Value: 2},
```

Comment:
The stepping functionality shunts values into the step bucket before the sample TS, so these tests all had to be skewed down by to an even numbered millisecond as the step size is 2ms. Let me know if you don't agree with this behaviour.

Reply:
Sounds good, but you could have made the step configurable in the tests and default to previous behaviour.